pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -683,19 +683,32 @@ func (r *ReconcileInfinispan) upgradeInfinispan(infinispan *infinispanv1.Infinis
 }
 
 func (r *ReconcileInfinispan) scheduleUpgradeIfNeeded(infinispan *infinispanv1.Infinispan, podList *corev1.PodList, logger logr.Logger) (*reconcile.Result, error) {
+	if upgrade, err := upgradeRequired(infinispan, podList); upgrade || err != nil {
+		if err := r.update(infinispan, func() {
+			logger.Info("schedule an Infinispan cluster upgrade", "pod default image", podList.Items[0].Spec.Containers[0], "desired image", consts.DefaultImageName)
```

Comment:
Probably image is missing, whole container will be printing

Reply:
I don't understand what you mean? In order for `upgradeRequired` to return true, it's necessary for `len(podList) > 0`.

Reply:
`podList.Items[0].Spec.Containers[0]` returns whole container object, not an image

Reply:
Gotcha! My copy and paste skills failed me there. Updated.